### PR TITLE
Fix: fix file structure and user deactivation

### PR DESCRIPTION
**Pull request by joboy-dev**
## Description

The changes made were to just update the way the files are structured and to include services to manage database queries and operations.
It also covers user deactivation and reactivation endpoints so as a user deactivates their account, they have a chance to reactivate the account after a set amount of time
​

## Motivation and Context

The changes are required so there is a structure in the codebase and to make the toutes a lot leaner which in turn makes the code a lot cleaner.
​

## How Has This Been Tested?

This has been tested using Postman extensively and captures all edge cases. It has also been tested with pytest covering all possible scenarios.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Dev-wonderful — DISMISSED** [review]

File: api/utils/success_response.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+def success_response(status_code: int, message: str, data: Optional[dict] = None):
+    '''Returns a JSOn response for success responses'''
+
+    response_data = {
+        "status_code": status_code,
+        "success": True,
+        "message": message
+    }
+
+    if data is not None:
```
Comment (zxenonx) [inline api/utils/success_response.py:14]:
```suggestion
    if data:
```

File: api/v1/models/newsletter.py [hunk 2]
```diff
@@ -2,15 +2,13 @@
 from uuid import uuid4
 from datetime import datetime
 from api.db.database import Base
+from api.v1.models.base_model import BaseTableModel
 
 
-class NEWSLETTER(Base):
+class NEWSLETTER(BaseTableModel):
```
Comment (zxenonx) [inline api/v1/models/newsletter.py:8]:
```suggestion
class Newsletter(BaseTableModel):
```

File: api/v1/schemas/user.py [hunk 3]
```diff
@@ -1,10 +1,63 @@
...
+import re
+
+class UserBase(BaseModel):
+    '''Base user schema'''
+
+    id: str
+    first_name: str
+    last_name: str
+    username: str
+    email: str
```
Comment (zxenonx) [inline api/v1/schemas/user.py:14]:
```suggestion
    email: EmailStr
```

File: api/v1/services/product.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    def fetch_all(self, db: Session, **query_params: Optional[Any]):
+        '''Fetch all products with option tto search using query parameters'''
+
+        query = db.query(Product)
+
+        # Enable filter by query parameter
+        if query_params:
+            for column, value in query_params.items():
+                if hasattr(Product, column) and value is not None:
```
Comment (zxenonx) [inline api/v1/services/product.py:31]:
```suggestion
                if hasattr(Product, column) and value:
```

if obj checks that obj is not empty/null (None)

File: tests/v1/user_deactivation_test.py [hunk 5]
```diff
@@ -97,7 +78,7 @@ def error_user_deactivation(test_db):
         "confirmation": False
     }, headers={'Authorization': f'Bearer {access_token}'})
     assert confirmation_false.status_code == 400
-    assert confirmation_false.json()['detail'] == 'Confirmation required to deactivate account'
+    assert confirmation_false.json()['message'] == 'Confirmation required to deactivate account'
 
 
     unauthorized = client.patch('/api/v1/users/accounts/deactivate', json={
```
Comment (zxenonx) [inline tests/v1/user_deactivation_test.py:84]: this route is not correct: /api/v1/users/accounts/deactivate

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @joboy-dev 
I left some comments

File: api/utils/success_response.py (hunk 1 above)
Comment (joboy-dev) [reply api/utils/success_response.py:14]: I'll change it now

File: tests/v1/user_deactivation_test.py (hunk 5 above)
Comment (joboy-dev) [reply tests/v1/user_deactivation_test.py:84]: The route is correct. It works fine
Or what do you mean please??

**Review by zxenonx — APPROVED** [review]
Thanks @joboy-dev

**Pull request merged**